src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -694,6 +694,26 @@ def gen_code(self, parent, options=None):
 
         self._add_call("PostEnd", parent)
 
+    def fix_gen_code(self, parent):
+        '''This function might be called from loop.gen_code if a PSyData node
```

Comment:
This will need updating if you can move your fix to LFRicLoop.

Reply:
Done.